eras/alonzo/impl/src/Cardano/Ledger/Alonzo/Tools.hs

```diff
@@ -87,6 +80,12 @@ data TransactionScriptFailure c
     NoCostModelInLedgerState !Language
   deriving (Show, Eq)
 
+-- | This 'Show' instance is intentionally an orphan,
+-- meant to be used only as part of 'TransactionScriptFailure'
+-- See the comment at the definition of 'PlutusDebug' about
+-- the cost of 'Show'
```

Comment:
Hmm, I am not sure what to do here. The show here is not in a testing module, and could therefore be abused.

Maybe we could require end-users to use a standalone function that we provide? What do you think @lehins ?

Reply:
Do we need to have `Show` instance for `TransactionScriptFailure`?

If we only need it for tests, then moving as standalone into a test package is the best approach IMHO

Reply:
The `Show` on `TransactionScriptFailure` itself is forcing it.

Reply:
That being said, if we really need this Show instance here or downstream, then adding one for `PlutusDebug` would be acceptable, we just need to be really sure not to use it. One way or another it should not be an orphan instance in this module

Reply:
> The Show on TransactionScriptFailure is forcing it.

I understand that, that's why I asked: "Do we need to have Show instance for TransactionScriptFailure?"

> Would it hurt if we removed it?

Try and remove it. See what breaks

Reply:
We could also ask @Jimbo4350 and @ak3n if they need a `Show` instance for `TransactionScriptFailure` (they are the two folks that I am aware of who use the tool).

Reply:
I'm pushing a version with the `Show` instance for both: `PlutusDebug` and `TransactionScriptFailure`, moved to the `Test` module, where those are used, for review.